### PR TITLE
New apis for Subscription, Addon, Plan, Virtual Account and bank transfer

### PR DESCRIPTION
**Pull request by hobbes09**
Edit operations added whereever applicable. Also checked that the new APIs have access under `private`

### CODE REVIEW

**Comment by hobbes09** [comment]
https://github.com/razorpay/api/issues/5647
https://github.com/razorpay/api/issues/5701

**Comment by pankajnits** [comment]
any visibility when it will be released

File: src/main/java/com/razorpay/AddonClient.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    /***
+     * To create an Addon, use the createAddon method of SubscriptionClient
+     */
+
+    public Addon fetch(String id) throws RazorpayException {
+        return get(String.format(Constants.ADDON_GET, id), null);
+    }
+
+    public Addon delete(String id) throws RazorpayException {
+        return get(String.format(Constants.ADDON_DELETE, id), null);
```
Comment (ankit-rzp) [inline src/main/java/com/razorpay/AddonClient.java]: why is delete a get call ?

File: src/main/java/com/razorpay/VirtualAccountClient.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public VirtualAccount close(String id) throws RazorpayException {
+
+        JSONObject request = new JSONObject();
+        request.put("status", "closed");
+
+        return patch(String.format(Constants.VIRTUAL_ACCOUNT_EDIT, id), request);
+    }
+
+    public List<Payment> fetchPayments(String id) throws RazorpayException {
+        return getCollection(String.format(Constants.VIRTUAL_ACCOUNT_PAYMENTS, id), new JSONObject());
```
Comment (ankit-rzp) [inline src/main/java/com/razorpay/VirtualAccountClient.java]: call `fetchPayments(String id, null)`

File: src/main/java/com/razorpay/ApiClient.java [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 import okhttp3.Response;
 
+import org.apache.commons.lang.WordUtils;
```
Comment (ankit-rzp) [inline src/main/java/com/razorpay/ApiClient.java]: `WordUtls` is not defined. Add the dependency in `pom.xml`

File: src/main/java/com/razorpay/ApiClient.java [hunk 4]
```diff
@@ -159,7 +166,7 @@ private void throwServerException(int statusCode, String responseBody) throws Ra
   private Class getClass(String entity) {
     try {
       String entityClass =
-          "com.razorpay." + Character.toUpperCase(entity.charAt(0)) + entity.substring(1);
+          "com.razorpay." + WordUtils.capitalize(entity, new char[]{'_'}).replaceAll("_", "");
```
Comment (ankit-rzp) [inline src/main/java/com/razorpay/ApiClient.java]: `WordUtils.capitalize(entity, '_')` will also work, as it takes variable length argument

File: src/main/java/com/razorpay/AddonClient.java (hunk 1 above)
Comment (hobbes09) [reply src/main/java/com/razorpay/AddonClient.java]: Seems like some mixup in last merge with testing branch ... corrected

File: src/main/java/com/razorpay/ApiClient.java (hunk 4 above)
Comment (hobbes09) [reply src/main/java/com/razorpay/ApiClient.java]: Second argument required is of type char[], whereas in this case, its char.

File: README.md [hunk 5]
```diff
@@ -275,3 +280,127 @@ Transfer transfer = razorpayClient.Transfers.fetch("transfer_id");
 ```java
 List<Transfer> transfers = razorpayClient.Transfers.fetchAll();
 ```
+
+### Subscriptions
```
Comment (ankit-rzp) [inline README.md]: add link to subscription doc

File: pom.xml [hunk 6]
```diff
@@ -63,6 +63,12 @@
       <artifactId>commons-codec</artifactId>
       <version>1.4</version>
     </dependency>
+
+    <dependency>
```
Comment (ankit-rzp) [inline pom.xml]: dependency is outdated.

**Pull request merged**